nixos/tests/memcached.nix

```diff
@@ -1,28 +1,24 @@
-import ./make-test.nix ({ pkgs, ...} : {
+import ./make-test-python.nix ({ pkgs, ... }: {
   name = "memcached";
 
-  nodes = {
-    machine =
-      { ... }:
-      {
-        imports = [ ../modules/profiles/minimal.nix ];
-        services.memcached.enable = true;
-      };
+  machine = {
+    imports = [ ../modules/profiles/minimal.nix ];
+    services.memcached.enable = true;
   };
 
   testScript = let
     testScript = pkgs.writeScript "testScript.py" ''
-      #!${pkgs.python3.withPackages (p: [p.memcached])}/bin/python
+      #!${pkgs.python3.withPackages (p: [ p.memcached ])}/bin/python
```

Comment:
how about using 

```nix
pkgs.writers.writePython3 { libraries = [ pkgs.python3Packages.memcached ]; } ''
  ...
'';

this will also run `flake8` analysis against the script.

Reply:
TIL about `writers`, thanks!